### PR TITLE
feat(login): check if the callback address is safe

### PR DESCRIPTION
**Pull request by nannan00**
### Description

- 检查登录回调地址是否安全

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/bk-login/bklogin/authentication/utils.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+# Copied from django.utils.http._url_has_allowed_host_and_scheme()
+# but additional support for wildcard domain matching.
+# 支持匹配:
+#  (1) * 匹配任意域名
+#  (2) 泛域名匹配，比如 .example.com 可匹配 foo.example.com、example.com、foo.example.com:8000、example.com:8080
+#  (3) 精确域名匹配，比如 example.com 可匹配 example.com、example.com:8000
+#  (4) 精确域名&端口匹配，比如 example.com:9000 只可匹配 example.com:9000
+def _url_has_allowed_host_and_scheme(url, allowed_hosts, require_https=False):
```
Comment (narasux) [inline src/bk-login/bklogin/authentication/utils.py:53]: Nit：适当加点空行

Comment (narasux) [reply src/bk-login/bklogin/authentication/utils.py:53]: 类型注解可以搞一手，上面那个函数也是

**Review by narasux — DISMISSED** [review]

File: src/bk-login/bklogin/authentication/utils.py (hunk 1 above)
Comment (nannan00) [reply src/bk-login/bklogin/authentication/utils.py]: DONE

File: src/bk-login/bklogin/authentication/utils.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+# Copied from django.utils.http._url_has_allowed_host_and_scheme()
+# but additional support for wildcard domain matching.
+# 支持匹配:
+#  (1) * 匹配任意域名
+#  (2) 泛域名匹配，比如 .example.com 可匹配 foo.example.com、example.com、foo.example.com:8000、example.com:8080
+#  (3) 精确域名匹配，比如 example.com 可匹配 example.com、example.com:8000
+#  (4) 精确域名&端口匹配，比如 example.com:9000 只可匹配 example.com:9000
+def _url_has_allowed_host_and_scheme(url: str, allowed_hosts: Set, require_https: bool = False):
```
Comment (narasux) [inline src/bk-login/bklogin/authentication/utils.py:59]: Set -> Set[str] ?  

allowed_hosts 好像不止支持 Set，从单元测试看 str 也可以？

**Review by narasux — APPROVED** [review]

**Pull request merged**